design/Abilities.md

```diff
@@ -0,0 +1,545 @@
+# Proposal: Abilities in Roc

```

Comment:
What do we think of adding a "status" here, to make it clear how (im)mature the implementation is? We can also link to implementation design from there if we think it's appropriate.

Reply:
Perhaps use multiple (checked) checkboxes to indicate progress and what's left to do.

Reply:
Personally, I feel that is better left to issue tracking - we already have that at https://github.com/roc-lang/roc/issues/2463. Maybe just something short that like "Status: we invite you to try out abilities for beta use, and are working on resolving known limitations"?